src/modsim2/similarity/mmd.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+from sklearn import metrics
+
+from modsim2.similarity.embeddings import EMBEDDING_FN_DICT
+
+MMD_KERNEL_DICT = {
+    "rbf": metrics.pairwise.rbf_kernel,
+    "laplace": metrics.pairwise.laplacian_kernel,
+}
+
+
+def mmd(
+    array_A: np.ndarray, array_B: np.ndarray, embedding_name: str, kernel_name: str
+) -> float:
+    """
+    Computes maximum mean discrepancy (MMD) for A and B. Given a kernel
+    embedding k(X,X'), computes
+
+        MMD^2 = 1/(N_A^2)*sum(k(A,A)) + 1/(N_B^2)*sum(k(B,B)) -
+                    2/(N_A * N_B)*sum(k(A,B))
+
+    Note that A and B must be matrices for the MMD. Currently mmd()
+    will reshape arbitarily size np.ndarrays to matrices. However,
+    other options include feature embeddings which will be explored in
+    the future.
+
+    Args:
+        array_A: The first image dataset
+        array_B: The second image dataset
+        kernel_name: An appropriate kernel embedding. See kernel_dict
+                     for choices
+
+    Returns:
+        float: The maximum mean discrepancy between A and B
+    """
+    # Extract embedding callable
+    embedding_fn = EMBEDDING_FN_DICT[embedding_name]
+
+    # In the future, consider optionally using feature embeddings here
+    matrix_A = embedding_fn(array_A)
+    matrix_B = embedding_fn(array_B)
+    N_A = matrix_A.shape[0]
+    N_B = matrix_B.shape[0]
+
+    # Extract kernel callable
+    kernel_fn = MMD_KERNEL_DICT[kernel_name]
+
+    # Compute MMD components
+    kernel_AA = kernel_fn(X=matrix_A)
```

Comment:
these ops tend to be expensive and looks like the sklearn ones don't provide their own logging. I think it'd be helpful to add log statements here (probably at INFO) to state we're about to compute kernel_fn [name] over N pairs of rows and that this may be a bit slow

Reply:
I've added this